src/www/ui/api/Controllers/CopyrightController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/*
+ SPDX-FileCopyrightText: © 2018 Siemens AG
+ Author: Gaurav Mishra <mishra.gaurav@siemens.com>
+ SPDX-FileCopyrightText: © 2023 Soham Banerjee <sohambanerjee4abc@hotmail.com>
+
+ SPDX-License-Identifier: GPL-2.0-only
+*/
+
+/**
+ * @file
+ * @brief Controller for copyright queries
+ */
+
+namespace Fossology\UI\Api\Controllers;
+
+use Fossology\Lib\Auth\Auth;
+use Fossology\Lib\Dao\UploadDao;
+use Fossology\Lib\Db\DbManager;
+use Fossology\UI\Api\Helper\ResponseHelper;
+use Fossology\UI\Api\Models\Info;
+use Fossology\UI\Api\Models\InfoType;
+use Psr\Http\Message\ServerRequestInterface;
+
+
+class CopyrightController extends RestController
+{
+  /**
+   * @var ContainerInterface $container
+   * Slim container
+   */
+  protected $container;
+
+  /**
+   * @var ClearingDao
+   */
+  private $clearingDao;
+
+  /**
+   * @var LicenseDao $licenseDao
+   * License Dao object
+   */
+  private $licenseDao;
+
+  /**
+   * @var UploadDao $uploadDao
+   * Upload Dao object
+   */
+  private $uploadDao;
+
+  /**
+   * @var CopyrightHist $CopyrightHist
+   * Copyright Histogram object
+   */
+  private $CopyrightHist;
+
+  public function __construct($container)
+  {
+    parent::__construct($container);
+    $this->container = $container;
+    $this->clearingDao = $this->container->get('dao.clearing');
+    $this->licenseDao = $this->container->get('dao.license');
+    $this->uploadDao = $container->get('dao.upload');
+    $this->restHelper = $container->get('helper.restHelper');
+    $this->CopyrightHist = $this->restHelper->getPlugin('ajax-copyright-hist');
+  }
+
+  /**
+   * Get all copyrights for a particular upload-tree
+   *
+   * @param  ServerRequestInterface $request
+   * @param  ResponseHelper         $response
+   * @param  array                  $args
+   * @return ResponseHelper
+   */
+  public function getTotalFileCopyrights($request, $response, $args)
+  {
+    $uploadPk = $args["id"];
+    $uploadTreeId = $args["itemId"];
+    $agentId = $this->CopyrightHist->getAgentId($uploadPk, 'copyright_ars');
+    $UploadTreeTableName = $this->uploadDao->getUploadtreeTableName($uploadPk);
+    $returnVal = null;
+    try {
+      if (!$this->dbHelper->doesIdExist("upload", "upload_pk", $uploadPk)) {
+        $returnVal = new Info(404, "Upload does not exist", InfoType::ERROR);
+      } else if (!$this->dbHelper->doesIdExist($this->uploadDao->getUploadtreeTableName($uploadPk), "uploadtree_pk", $uploadTreeId)) {
+        $returnVal = new Info(404, "Item does not exist", InfoType::ERROR);
+      }
+      if ($returnVal !== null) {
+        return $response->withJson($returnVal->getArray(), $returnVal->getCode());
+      }
+      $returnVal = $this->CopyrightHist->getCopyrights($uploadPk, $uploadTreeId, $UploadTreeTableName, $agentId, 'statement', 'active', true);
+      $finalValue = array('total_copyrights' => $returnVal[2]);
+      return $response->withJson($finalValue, 200);
```

Comment:
Please use more efficient ways:
```suggestion
      return $response->withJson(["total_copyrights" => $returnVal[2]], 200);
```

Reply:
Even faster way is to move following lines to CopyrightDao as the function `getCopyrights()` also fetches other things from DB which are not required for this endpoint.

https://github.com/fossology/fossology/blob/fc89e1f3153d7c6eb088942ef6f0a2b44b58ade2/src/copyright/ui/ajax-copyright-hist.php#L288-L292